issue-40-built-in-benchmarking-tool.md

```diff
@@ -0,0 +1,35 @@
+## Add Built-in Benchmarking Tool
+
+https://github.com/submariner-io/submariner-website/issues/40
+
+## Summary
+
+This is a proposal to support a built-in benchmarking tool in Submariner.
+The goal is to provide users with easy way to benchmark the network performance across clusters, and experiment with different tuning of the system.
+
+
+## Proposal
+
+Add a new ability to have a built-in benchmark tool.
+Implement a new Subctl command that will use iperf3 internally to benchmark the throughput between two nodes.
+
+scheduling server and client can be placed within the same cluster or spread across multiple clusters.
+
+The user can specify source and destination nodes via node names. (Or via lables?)
+
+If nodes are not specified, the following tests will be run by default:
+* non-gateway to non-gateway  
+* gateway to gateway
```

Comment:
@skitt: Based on the ODCN experience, do we want to give an option to the user to schedule pods with HostNetworking?

@nyechiel @mkolesnik @maayanf24 thoughts?

Reply:
As I mentioned before, to me it seems that the aim of this tool is to measure inter-cluster traffic in order to test and compare different configurations of submariner (or perhaps you could compare it with other solutions, though I'm not sure that's something we'd like to support or not).

As such, I argue that the best path forward is to give a simple set of tests that would be performed by the tool and reported so they can be easily compared.

We need to decide which metrics make sense and which deployments make sense to test.

Obviously this list can be further expanded, but I feel the tool should remain simple and intuitive and give clear (and few) options to the user to facilitate ease of use.

We're not trying to build "the ultimate K8s network benchmarking tool" here, so I think we should keep it simple and focus on the things that give the most value to the most people.

Reply:
I agree with @mkolesnik here on providing a set of standard tests which would let people compare deployment to deployment easily.. may be we can add a selector for specific parts (for example if one path is being optimized and they need to re-run a specific bit, but I'd say that's a later enhancement).

Reply:
Okay, for now, we can skip this and we can consider this as a future item.